### ISSUE TITLE
"An error occurred while attempting to link Alexa" - how to debug?

### ISSUE THREAD

**Issue by cloudon7281** [alexa/alexa-smarthome#24]
Hi

I've been following the 15 minute tutorial at https://github.com/alexa/alexa-smarthome/wiki/Build-a-Working-Smart-Home-Skill-in-15-Minutes, and am hitting an error at the linking stage.

So far, I've
- set up LWA as per https://developer.amazon.com/blogs/alexa/post/Tx4WG410EHXIYQ/five-steps-before-developing-a-smart-home-skill
- created the skill
- created the lambda function
- configured the skill [note: trying to do this with a EU (Ireland) region lambda failed; I had to do it with a US-East-1 - is that a known restriction?]

It's then going wrong when I try to enable the skill.  I go to https://alexa.amazon.com and can find the skill under Your Skills with the devUS tag.  I then get a "click 'Allow ' to sign in to..." dialog, and when I click allow I get "an error occurred while attempting to link Alexa to xxx - please try again later".  I've tried this several times at different times with same result.

Any clue?  Nothing is showing up in the lambda logs, not that I'd expect it to.

thanks

Calum

**Comment by briankel** [comment]
Please check your CloudWatch logs for any clues.
Please also make sure you added the redirect URI for your authentication provider. (in the instructions see "Provide Redirect URL's to LWA:"

Regarding your other question, Alexa is only supported as a trigger in specific AWS regions. For Europe you should be able to make this work in eu-west-1.

**Comment by cloudon7281** [comment]
Thanks for the reply.

I believe I have the redirect URIs set up correctly: under Developer Console->Apps & Services->Security Profiles-><my profile>->Web Settings then I have three allowed return URLs:

https://layla.amazon.com/api/skill/link/<code>
https://pitangui.amazon.com/api/skill/link/<code>
https://alexa.amazon.co.jp/api/skill/link/<code>

which match those configured for the redirect URLs under Developer Console->Alexa-><my skill>->Configuration.

Cloudwatch is enabled for my lambda, but is not showing any attempted accesses, and there are no traces showing up in X-Ray.  I can't see any way to add Cloudwatch logs to my skill.  Is there a way to do so?

**Comment by briankel** [comment]
Hmm...
No, you can't add CloudWatch to your skill specifically - just to the Lambda.
I don't have any immediate ideas for you, just to go back to the 15-minute skill instructions and make sure you've followed each step exactly. There are a lot of steps and it's easy to miss one. If I think of other ideas I'll post them back here.

**Comment by itssindhu13** [comment]
HI Calum,

Were you able to fix this issue you had? Facing with the same and came across this ticket. Appreciate any help.

Thanks
Sindhu

**Comment by muenzpraeger** [comment]
Same issue here @briankel.

**Comment by cloudon7281** [comment]
Still no success.  I wondered if it might be down to the issues raised in https://github.com/alexa/alexa-smarthome/issues/28 so went through it again with an English (UK) skill and an EU (Ireland) lambda, but precisely the same error is occurring.

**Comment by itssindhu13** [comment]
For me, the issue was that I used Python 2.7 instead of Python 3.6 in Lambda. That change fixed it for me.

**Comment by cloudon7281** [comment]
OK, solved: I'm an idiot.  I had an incorrect lambda handler function specified (lambda_function.lambda_handler, not lambda.lambda_handler), which I found when I tried explicitly testing my lambda.

@briankel Suggestion for the documentation: at the end of the "Create the Lambda function" section, before "Configure skill", I suggest adding a step of explicitly testing your lambda to check it has been correctly configured.
